### ISSUE TITLE
Clarify in the Message tooltip that %i stops messages from being printed in chat

### ISSUE THREAD

**Issue by Wisdawn** [WeakAuras/WeakAuras2#3244]
**Is your feature request related to a problem? Please describe.**
Yes, this is a problem that can lead to confusion and unnecessary support requests. Adding an icon to a Chat Message via %i stops the message from being printed in any chat channel, thus stops the announcement part of a WeakAura without a clear reason why the announcement is not working.

**Describe the solution you'd like**
I think it would be really worth it to add a note about this in the tooltip titled "Message," which shows up when mousing over the input field for the chat message, because {skull}, for example, works in chat just fine, turning itself into an inline skull marker in chat messages, so this will probably give the impression to some users that spell icons too can or should work fine in chat messages.

**Describe alternatives you've considered**
I've already used the alternative of submitting my aura to Wago and getting help from the Discord #retail-auras-help, and thanks to nullKomplex, I learned from them that %i prevents chat messages from being printed. I think it would be much better to have a simple short sentence in the tooltip clarifying this issue.

**Additional context**
None, thank you.

**Comment by asaka-wa** [comment]
Tbh, if WA is firing SendChatMessage for you then it should probably just strip any formatting from it that can't be sent. 
Feels a bit niche for a warning message.

**Comment by Wisdawn** [comment]
That is certainly a valid solution, but it would require a bit of coding, rather than simply adding a sentence to an existing tooltip. Moreover, it might still lead some users to come to ask for support as to why their icons are not showing up in the chat messages printed by WA. 

Alternatively, perhaps the %i could be removed altogether from that tooltip, so that users who aren't aware of it won't know about it and try to use it?

Of course, I don't know if this is a common tooltip used for purposes other than the chat message field, but if it cannot be used in a chat message, might as well be removed from that specific tooltip.

**Comment by InfusOnWoW** [comment]
After looking at that for a few minutes, I think there's very little benefit to a warning message. 

It explains a niche case that pratically no one will ever encounter. In fact as far as I know this is very first time it has ever come up in the decade that this has been the behaviour.

And we have experience with the existing warning message for Say/Yell not working outside of instances. That warning gets overlooked all the time.

That makes the warning message have very little benefit. And the cost is a worse ui for all people that are not interested in this.

**Comment by Wisdawn** [comment]
Worse UI because of including extra text? 

Well, I mentioned an alternative in my response above, and I can rephrase the alternative as a confused exclamation: Why is the tooltip mentioning the %i _at all_ if it cannot actually be used in that text field?! Please, just remove it.

If you think adding text makes for a worse UI, then surely removing text makes for a simpler, better UI. So, just remove the %i altogether from that specific tooltip. It is clearly misleading, regardless of whether someone will use it or not. 

And I'm willing to bet that in the decade that this has been the behavior this is the first time this came up simply because I'm one of the very, very rare users who care enough to report issues and understand root causes. The vast majority of users, or humans generally, just ignore whatever isn't working and give up on it without providing feedback or reporting bugs. So, it's not like I'm the only one who encountered this problem in 10 years; I'm the only one who simply reported it, instead of just silently thought to myself that scripting WA is too complex for me to figure out.

**Comment by InfusOnWoW** [comment]
Yes it's a worse UI for the 99.999% of all users that the text is not relevant too. 

The problem is not specific to %i at all, it can happen with every text replacement that returns a ui escape sequence. E.g. the class color formating we added for %unit. Thus the warning text would be: "Chat does not support UI escape sequences". If that's helpful to anyone, I'd be very suprised.

Well, you didn't report it as a issue. You made a feature suggestion.

**Comment by Wisdawn** [comment]
If references other than %i cannot be sent to chat lines, then those references should be removed too from the tooltip that explains to new users what they can and cannot add to their chat lines! Why even include a reference in a tooltip where the reference cannot be used?!

I am confused as to how you don't clearly see a problem in the addon's current reference material, tooltips constituting a main part of which, since users who read tooltips are astronomically more than users who Alt-tab to read Wikis or the like.

A user mouses over a text field that they can put their custom text in, and a tooltip pops up giving them — supposedly — helpful information about what references they can include in their custom chat lines, the emphasis here being on "_can_," so if a user cannot use any number of references because they stop the whole text from being printed in a chat channel, then those references should most definitely be removed from the tooltip that shows up when mousing over that specific text field. The current tooltip is, literally, misleading users. And it sounds like you're caring a lot more about the 99.9999% of the users who don't read tooltips, and are quite happy to keep misleading the users who do read tooltips and are trying to experiment with the addon or see what they _can_ do with it.

Even one of your own volunteers who are trying to help answer people's questions couldn't figure out the problem caused by this issue, and another volunteer had to get involved. And that volunteer, who did figure out the issue, was so surprised that a user tried to include the spell's icon in a chat line about that spell! Imagine that ... someone finding the use of a spell's icon in the chat line about that spell weird, or strange, or very mysterious as to what it's trying to achieve. It's like, "Hmm ... there's a spell in the game, and that spell ... not surprisingly ... has an icon, just like 100% of the spells of all graphical games in the history of gaming ... now, why would anyone in their right mind use a spell's icon when trying to send a warning about that spell?! What are they trying to achieve by using the icon? Hmm.  Strange. Even DBM doesn't do that in warnings or anything." Yeah, very strange. Just as strange as the fact that 100% of spells in graphical gaming history have icons. And as strange as the fact that humans can recognize a visual icon astronomically faster than they can read the name of the icon, which can help humans ignore recurring messages if they see a familiar icon on them and are already familiar with the message.

Regarding whether this is an "issue" or not, please take a look up at the top of this page, and find under which tab exactly this issue is posted. There is no "Suggestions" tab; there's just the Issues tab. And you only have two sub-categories under Issues: Bugs or Suggestions. This issue is not a "bug" because it doesn't stop your own addon's code from working, but it is most definitely an issue, as I tried over and over again above to demonstrate. It's an issue that is misleading the users who are trying to learn your complex piece of software, and an issue that some of your own support volunteers cannot even figure out sometimes, and the only reason they're not getting more queries about it is that very few people are going to care enough to ask about it.

And by the way, automated /say and /yell is a regular issue, not because people ignore the warning you've included in the addon, it's a regular issue because tons of users who have never put together a single aura in their life download Wago auras to announce things, and the auras they download tell them that they're supposed to be announcing certain things in /say and /yell, and /say and /yell work in-game outside instances just fine, so when their downloaded auras don't work, they come to your Discord, occasionally, to ask what's going on. And I know you get so many queries for this issue, but I say "occasionally" because you truly are getting only a small fraction of the total users who have tried those auras, found them "not working as expected," and never bothered to even look for an answer as to why they're not working. 

So, the "/say and /yell in instances only" issue is not an issue with people ignoring your tooltips or reference material; it's an issue with people downloading auras that advertise a result, and that result is expected to work in a certain way because that's how it always worked that way, and it still does when it comes to things other than automated announcements by WA. I understand why you chose to disallow automated /say and /yell by WeakAuras, but I invite you to instead of thinking that people are ignoring your reference material, realize that you made a certain imperfect decision that confuses some of the Wago WA users, and for good reason. So perhaps you can make a meeting about finding a solution for this issue that keeps leeching on your volunteers' time, and probably makes them roll their eyes every time someone new comes asking about it. Maybe WA can detect when an aura is trying to print a chat line to /say or /yell outside an instance, and simply print an error message to the user of the aura themselves clarifying that WA is not going to print in /yell or /say channels outside an instance. That, alone, would probably reduce the amount of confusion and repetitive queries greatly.

Maybe WA could've also printed an error message for me telling me that the chat line in the aura could not be printed because it includes an invalid reference, which is %i. If I saw such an error message, I could've removed this reference, tried the aura again, found it working, and then I would've been confused why the %i was in that tooltip at all if it cannot be used, and why WA's tooltips are this misleading. And believe me, I would've never come here or on Discord, then, to endeavor to help you see that your tooltip is misleading ... and clearly failing.

**Comment by InfusOnWoW** [comment]
For the record, you choose to make this a feature suggestion not a bug report. Apparently you now think this is a bug report. Whatever.

As to SAY/YELL, the limitation is not in WeakAuras, but in WoW. 

You didn't understand my point about class-colored %unit at all. In general there's no way to figure out whether a given text replacement will contain a UI escape sequence or not. Thus there is no general way to detect if a given text replacement will lead to a unprintable output, and thus a generic warning message would need to show anytime the user selects a chat output channel. 

I did consider various other approaches, thus as parsing the text for escape sequences and other disallowed codepoints. Or escaping the escape sequences.

I have for judged with all that in mind, that for now I choose to do nothing about it. Maybe that's wrong, maybe it's the right thing to do. My job as a maintainer is to figure out what the right approach is, and I have made my decission. There's little point in further discussions.

WeakAuras, in lots of places is close, to the underlying wow api and this is simply a place where that shows through. The volunteers did not know about this limitation, because it has probably not come up before.
